lib/server/services/dataPasses/DataPassService.js

```diff
@@ -19,13 +19,27 @@ const { DataPassVersionRepository, GaqDetectorRepository, RunRepository, DplDete
 const { getOneDataPass } = require('./getOneDataPass.js');
 const { getOneDataPassOrFail } = require('./getOneDataPassOrFail.js');
 const { BadParameterError } = require('../../errors/BadParameterError');
+const { AccessDeniedError } = require('../../errors/AccessDeniedError');
+const { BkpRoles } = require('../../../domain/enums/BkpRoles');
 
 /**
  * @typedef DataPassIdentifier
  * @param {number} [id]
  * @param {string} [name]
  */
 
+/**
+ * Check whether given user's roles suffice to manage GAQ detectors
+ * @param {string[]} userRoles roles of user
+ * @return {void}
+ * @throws {AccessDeniedError} when roles don't suffice
+ */
+const validateUserGaqAccess = (userRoles) => {
```

Comment:
I do not think there is a need for this check here.
Did you have a look at `rbacMiddleware`? For usage check `tags.router.js`

Reply:
applied rbac